templates/_db-migrate.tpl

```diff
@@ -107,7 +107,7 @@ spec:
                   {{- else }}
                   name: {{ template "mastodon.redis.secretName" . }}
                   {{- end }}
-                  key: redis-password
+                  key: {{ .Values.redis.auth.existingSecretKey }}
```

Comment:
I think it would be better to have both `name` and `key` (the whole `secretKeyRef` be from a template, rather than repeating this everywhere?

Reply:
That's honestly not a bad idea. I'll make that change.

Reply:
Except I can see one issue with this.

This solution allows it to be backwards compatible with the way it's already done. This would basically mean either replacing `existingSecret` with that new format, which would break existing setups, or creating a new field like `existingSecretRef`, which wouldn't break anything, but would be somewhat redundant.